SU2_CFD/src/drivers/CDriver.cpp

```diff
@@ -58,6 +49,48 @@
 #include "../../include/solvers/CDiscAdjFEASolver.hpp"
 #include "../../include/solvers/CFEM_DG_EulerSolver.hpp"
 #include "../../include/solvers/CFEM_DG_NSSolver.hpp"
+#include "../../include/solvers/CDiscAdjMeshSolver.hpp"
+#include "../../include/solvers/CMeshSolver.hpp"
+
+#include "../../include/interfaces/cfd/CConservativeVarsInterface.hpp"
+#include "../../include/interfaces/cfd/CMixingPlaneInterface.hpp"
+#include "../../include/interfaces/cfd/CSlidingInterface.hpp"
+#include "../../include/interfaces/cht/CConjugateHeatInterface.hpp"
+#include "../../include/interfaces/fsi/CDisplacementsInterface.hpp"
+#include "../../include/interfaces/fsi/CFlowTractionInterface.hpp"
+#include "../../include/interfaces/fsi/CDiscAdjFlowTractionInterface.hpp"
+#include "../../include/interfaces/fsi/CDisplacementsInterfaceLegacy.hpp"
+#include "../../include/interfaces/fsi/CDiscAdjDisplacementsInterfaceLegacy.hpp"
+
+#include "../../include/numerics/template.hpp"
+#include "../../include/numerics/transition.hpp"
+#include "../../include/numerics/heat.hpp"
+#include "../../include/numerics/flow/convection/roe.hpp"
+#include "../../include/numerics/flow/convection/fds.hpp"
+#include "../../include/numerics/flow/convection/fvs.hpp"
+#include "../../include/numerics/flow/convection/cusp.hpp"
+#include "../../include/numerics/flow/convection/hllc.hpp"
+#include "../../include/numerics/flow/convection/ausm_slau.hpp"
+#include "../../include/numerics/flow/convection/centered.hpp"
+#include "../../include/numerics/flow/flow_diffusion.hpp"
+#include "../../include/numerics/flow/flow_sources.hpp"
+#include "../../include/numerics/continuous_adjoint/adj_convection.hpp"
+#include "../../include/numerics/continuous_adjoint/adj_diffusion.hpp"
+#include "../../include/numerics/continuous_adjoint/adj_sources.hpp"
+#include "../../include/numerics/turbulent/turb_convection.hpp"
+#include "../../include/numerics/turbulent/turb_diffusion.hpp"
+#include "../../include/numerics/turbulent/turb_sources.hpp"
+#include "../../include/numerics/elasticity/CFEAElasticity.hpp"
+#include "../../include/numerics/elasticity/CFEALinearElasticity.hpp"
+#include "../../include/numerics/elasticity/CFEANonlinearElasticity.hpp"
+#include "../../include/numerics/elasticity/nonlinear_models.hpp"
+
+#include "../../include/integration/CSingleGridIntegration.hpp"
+#include "../../include/integration/CMultiGridIntegration.hpp"
+#include "../../include/integration/CStructuralIntegration.hpp"
+#include "../../include/integration/CFEM_DG_Integration.hpp"
```

Comment:
The integration has been restructured into these files, I also refactored the classes a bit (lot) moving methods from the parent to the children when appropriate and making things private/protected to reduce the public interface of CIntegration as much as possible.